### PR TITLE
[CALCITE-2588] Run Geode Adapter tests with an Embedded Instance

### PR DESCRIPTION
**Pull request by asereda-gs**
Part of the effort to increase test coverage for adapters. Enables execution of all Geode adapter tests during CI build (instead of manual run).

Create special JUnit Rule (`GeodeEmbeddedPolicy`) which starts / stops embedded Geode instance once for all the tests.

Modify GeodeSchema and GeodeTable to accept exising geode instance in the constructor instance of connection parameters.

Ignore two tests which currently fail for both embedded and external instance(s). These bugs will be addressed separately

Local test dataset is taken from `calcite-test-dataset` project (the one which is used for integration testing). Re-use `zips-mini.json` from mongo/elastic adapters.

### CODE REVIEW

File: geode/src/test/java/org/apache/calcite/adapter/geode/rel/JsonLoader.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  }
+
+  void loadClasspathResource(String location) throws IOException {
+    Objects.requireNonNull(location, "location");
+    InputStream is = getClass().getResourceAsStream(location);
+    if (is == null) {
+      throw new IllegalArgumentException("Resource " + location + " not found in the classpath");
+    }
+
+    load(new InputStreamReader(is, Charset.defaultCharset()));
```
Comment (vlsi) [inline geode/src/test/java/org/apache/calcite/adapter/geode/rel/JsonLoader.java]: Would you please use UTF-8 explicitly?

File: geode/src/test/java/org/apache/calcite/adapter/geode/rel/GeodeZipsTest.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.Collections;
+
+/**
+ * Tests based on {@code zips-min.json} dataset. Runs automatically as part of CI.
+ */
+public class GeodeZipsTest extends AbstractGeodeTest {
```
Comment (vlsi) [inline geode/src/test/java/org/apache/calcite/adapter/geode/rel/GeodeZipsTest.java]: Just wondering: should the test be included to a suite?

File: geode/src/test/java/org/apache/calcite/adapter/geode/rel/JsonLoader.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  }
+
+  void loadClasspathResource(String location) throws IOException {
+    Objects.requireNonNull(location, "location");
+    InputStream is = getClass().getResourceAsStream(location);
+    if (is == null) {
+      throw new IllegalArgumentException("Resource " + location + " not found in the classpath");
+    }
+
+    load(new InputStreamReader(is, Charsets.UTF_8));
```
Comment (vlsi) [inline geode/src/test/java/org/apache/calcite/adapter/geode/rel/JsonLoader.java]: Please use java.nio.charset.StandardCharsets#UTF_8

File: geode/src/test/java/org/apache/calcite/adapter/geode/rel/GeodeZipsTest.java (hunk 2 above)
Comment (asereda-gs) [reply geode/src/test/java/org/apache/calcite/adapter/geode/rel/GeodeZipsTest.java]: You want to create something like GeodeSuite ? What is the advantage (they will be discovered automatically) ?

Comment (vlsi) [reply geode/src/test/java/org/apache/calcite/adapter/geode/rel/GeodeZipsTest.java]: Geode is not using suites and that is great.

Comment (asereda-gs) [reply geode/src/test/java/org/apache/calcite/adapter/geode/rel/GeodeZipsTest.java]: So are you OK with leaving things as is ? Or do you want to change something about `AbstractGeodeTest`?

Comment (vlsi) [reply geode/src/test/java/org/apache/calcite/adapter/geode/rel/GeodeZipsTest.java]: I'm ok with this provided the test runs during default `mvn verify` execution

**Comment by vlsi** [comment]
@asereda-gs , Geode tests print a bunch of SQL to the console: https://travis-ci.org/apache/calcite/jobs/432201677#L3395

Would you please disable that output by default? (see https://github.com/apache/calcite/commit/3183f2cae6db91f0c2bb0fc99ff6121da56652b8 )

**Pull request closed without merging**